src/Package/Impl/Sql/Publish/SProcScriptGenerator.cs

```diff
@@ -0,0 +1,135 @@
+                if (i < sprocMap.Count - 1) {
+                    sb.Append(',');
+                }
+                sb.AppendLine(string.Empty);
+                i++;
+            }
+            return sb.ToString();
+        }
+
+        /// <summary>
+        /// Creates code for stored procedures
+        /// </summary>
+        public IReadOnlyDictionary<string, string> CreateStoredProcedureScripts(SqlSProcPublishSettings settings, IEnumerable<string> sprocFiles) {
+            var dict = new Dictionary<string, string>();
+            foreach (var rFilePath in sprocFiles) {
+                var sprocName = _fs.GetSProcNameFromTemplate(rFilePath);
+                if (!string.IsNullOrEmpty(sprocName)) {
+                    string template;
+                    if (settings.CodePlacement == RCodePlacement.Inline) {
+                        template = FillSprocInlineTemplate(rFilePath, sprocName);
+                    } else {
+                        template = FillSprocTableTemplate(rFilePath, sprocName, settings.TableName, settings.QuoteType);
+                    }
+                    if (!string.IsNullOrEmpty(template)) {
+                        dict[sprocName] = template;
+                    }
+                }
+            }
+            return dict;
+        }
+
+        /// <summary>
+        /// Replaces procedure name, R Code and the SQL query placeholders with actual values
+        /// </summary>
+        private string FillSprocInlineTemplate(string rFilePath, string sprocName) {
+            var sprocTemplateFile = rFilePath.ToSProcFilePath();
+            var sprocTemplate = GetSqlFileContent(sprocTemplateFile);
+
+            var rCode = GetRFileContent(rFilePath);
+            rCode = rCode.EndsWithOrdinal(Environment.NewLine) ? rCode : rCode + Environment.NewLine;
+            sprocTemplate = sprocTemplate.Replace(RCodeTemplate, Environment.NewLine + rCode);
+
+            var sqlQuery = GetSqlFileContent(rFilePath.ToQueryFilePath()).Trim();
+            return sprocTemplate.Replace(InputQueryTemplate, sqlQuery);
+        }
+
+        private string FillSprocTableTemplate(string rFilePath, string sprocName, string codeTableName, SqlQuoteType quoteType) {
+            var sprocTemplateFile = rFilePath.ToSProcFilePath();
+            var sprocTemplate = GetSqlFileContent(sprocTemplateFile);
+
+            var format =
+@"BEGIN
+DECLARE @RCodeQuery NVARCHAR(max);
+DECLARE @RCode NVARCHAR(max);
+DECLARE @ParmDefinition NVARCHAR(max);
+
+SET @RCodeQuery = N'SELECT @RCodeOUT = RCode FROM {0} WHERE SProcName = ''{1}''';
+SET @ParmDefinition = N'@RCodeOUT NVARCHAR(max) OUTPUT';
+
+EXEC sp_executesql @RCodeQuery, @ParmDefinition, @RCodeOUT=@RCode OUTPUT;
+SELECT @RCode;
+";
+            var declarations = string.Format(CultureInfo.InvariantCulture, format, codeTableName.ToSqlName(quoteType), sprocName);
+            sprocTemplate = sprocTemplate.Replace("BEGIN", declarations);
```

Comment:
I hope not :-) I am not T-SQL expert really but from what I see if there are multiple statements, there has to be ```GO``` between them. Since template intent to have code for a single procedure there shouldn't be more unless user types more of them. Not sure what would be the useful scenario though.